flang/docs/Extensions.md

```diff
@@ -160,7 +160,10 @@ end
 * `<>` as synonym for `.NE.` and `/=`
 * `$` and `@` as legal characters in names
 * Initialization in type declaration statements using `/values/`
-* Saved variables without explicit or default initializers are zero initialized.
+* Saved variables without explicit or default initializers are zero initialized,
+  except for implicitly saved scalar variables from the main program that are
```

Comment:
it might be more clear to the general reader to replace "implicitly saved scalar variables from the main program" with "scalar variables from the main program that are not explicitly initialized or marked with an explicit `SAVE` attribute".

I still think that somebody should add an option to the flang-new driver to cause main program variables to be static; the hooks are there in Fortran-features.h to do so.

Reply:
I can add the driver option later this week.

Reply:
Thanks @kiranchandramohan, I actually started a patch before seeing your comment: https://github.com/llvm/llvm-project/pull/121968